modules/server/src/network/index.ts

```diff
@@ -1,10 +1,111 @@
+ * @param input
+ * @returns
+ */
+const isGqlIntrospectionQuery = (input: unknown): input is IntrospectionQuery =>
+	!!input && typeof input === 'object' && '__schema' in input && typeof input.__schema === 'object';
+
+type FetchRemoteSchemaResult = {
+	config: NetworkAggregationConfigInput;
+	introspectionResult: IntrospectionQuery | null;
+};
+
+/**
+ *
+ * @param config - network config from env
+ * @returns a promise containing network config and the introspection query result
+ *
+ * @throws Fetch failed error
+ *
+ * @throws JSON parse error
+ *
+ * @throws Unexpected data error eg. Not a valid introspectionQuery result
+ */
+const fetchRemoteSchema = async (
+	config: NetworkAggregationConfigInput,
+): Promise<FetchRemoteSchemaResult> => {
+	const { graphqlUrl } = config;
+	try {
+		/**
+		 * get full schema (needed for buildClientSchema) and convert json
+		 */
+		const response = await fetchGql({
+			url: graphqlUrl,
+			gqlRequest: { query: getIntrospectionQuery() },
+		});
+
+		if (response.status === 200 && response.statusText === 'OK') {
+			// axios response "data" field, graphql response "data" field
+			const responseData = response.data?.data;
+
+			if (isGqlIntrospectionQuery(responseData)) {
+				return { config, introspectionResult: responseData };
+			} else {
+				console.error('unexpected response data in fetchRemoteSchema');
+				throw Error('unexpected response data');
+			}
+		} else {
+			console.error('network error in fetchRemoteSchema');
+			throw Error('network error');
+		}
+	} catch (error) {
+		console.log(`failed to retrieve schema from url: ${config.graphqlUrl}`);
+		console.error(error);
+		return { config, introspectionResult: null };
+	}
+};
+
+const fetchRemoteSchemas = async ({
 	networkConfig,
 }: {
-	networkConfig: NetworkAggregationInterface[];
+	networkConfig: NetworkAggregationConfigInput[];
+}): Promise<NetworkAggregationConfig[]> => {
+	// query remote notes
+	const networkQueryPromises = networkConfig.map(async (config) => fetchRemoteSchema(config));
+	// type cast because rejected errors are handled
+	const networkQueries = (await Promise.allSettled(
+		networkQueryPromises,
+	)) as PromiseFulfilledResult<FetchRemoteSchemaResult>[];
+
+	// build schema
+	const schemaResults = networkQueries.map((networkResult) => {
+		const { config, introspectionResult } = networkResult.value;
+
+		try {
+			const schema = introspectionResult !== null ? buildClientSchema(introspectionResult) : null;
+			return { ...config, schema };
+		} catch (error) {
+			console.error('build schema error', error);
+			return { ...config, schema: null };
+		}
+	});
+
+	return schemaResults;
+};
+
+export const createSchemaFromNetworkConfig = async ({
+	networkConfig,
+}: {
+	networkConfig: NetworkAggregationConfigInput[];
 }) => {
-	console.log('network config', networkConfig);
+	const remoteSchemasResult = await fetchRemoteSchemas({
+		networkConfig,
+	});
+
+	const mergedFields = mergeRemoteSchemas(remoteSchemasResult);
+};
```

Comment:
No return statement? When I know what type a function needs to return I like to specify this in the signature. This does two things:

1. TS will enforce that your returns are the correct type
2. Future devs can see at a glance what the function is supposed to return.

I think a function named "createXYZConfig" is a good candidate for this, it should state:

```ts
export const createSchemaFromNetworkConfig = async(inputs: {networkConfigs: NetworkAggregationConfigInput[] }): NetworkSearchSchema => {

```